add-on/src/lib/ipfs-companion.js

```diff
@@ -28,6 +36,29 @@ function initIpfsApi (ipfsApiUrl) {
   return window.IpfsApi({host: url.hostname, port: url.port, procotol: url.protocol})
 }
 
+async function getIpfsConfig (ipfs) {
+  try {
+    const buff = await ipfs.config.get()
+    return JSON.parse(buff.toString())
+  } catch (err) {
+    console.log('Failed to get IPSF config', err)
+    return null
+  }
+}
+
+async function getGatewayUrl (ipfs) {
+  const config = await getIpfsConfig(ipfs)
```

Comment:
Can you just fetch the `Addresses.Gateway` value?

```sh
$ ipfs config 'Addresses.Gateway'
/ip4/127.0.0.1/tcp/8080
```

Reply:
Sure can. I thought that getting the entire ipfs config might provide other useful info. Perhaps we won't ever need it, but I don't think it's a bad gamble.